# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1243364266 14400
#      Tue May 26 14:57:46 2009 -0400
# Node ID 7f3a1a90f4f48ea3ac3e93ac216d29f6ad328751
# Parent  be7d8132c1392a90ce12f11c1ad7182eb0c9f648
imread: scale values from old Octave image format

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2009-05-26  John W. Eaton  <jwe@octave.org>
+
+	* image/imread.m: Scale image data from old Octave image format so
+	that it will work with imshow.
+	* deprecated/loadimage.m: Undo scaling here for compatibility.
+
 2009-05-26 Søren Hauberg  <hauberg@gmail.com>
 
 	* help/__makeinfo__.m: Support several @seealso's in one text. 
 
 2009-05-26  John W. Eaton  <jwe@octave.org>
 
 	* plot/colorbar.m: Downcase location argument.
 
diff --git a/scripts/deprecated/loadimage.m b/scripts/deprecated/loadimage.m
--- a/scripts/deprecated/loadimage.m
+++ b/scripts/deprecated/loadimage.m
@@ -36,9 +36,11 @@ function [img_retval, map_retval] = load
   if (! warned)
     warned = true;
     warning ("Octave:deprecated-function",
              "loadimage is obsolete and will be removed from a future version of Octave; please use imread instead");
   endif
 
   [img_retval, map_retval] = imread (varargin{:});
 
+  img_retval *= 64;
+
 endfunction
diff --git a/scripts/image/imread.m b/scripts/image/imread.m
--- a/scripts/image/imread.m
+++ b/scripts/image/imread.m
@@ -72,19 +72,19 @@ function varargout = imread (filename, v
       endif
     catch
       error ("imread: invalid image file: %s", magick_error);
     end_try_catch
 
     if (map_field && (img_field || x_field))
       varargout{2} = vars.map;
       if (img_field)
-	varargout{1} = vars.img;
+	varargout{1} = vars.img / 64;
       else
-	varargout{1} = vars.X;
+	varargout{1} = vars.X / 64;
       endif
     else
       error ("imread: invalid Octave image file format");
     endif
 
   end_try_catch
 
 endfunction
